server/src/main/java/org/apache/druid/server/coordinator/duty/CompactSegments.java

```diff
@@ -154,6 +160,17 @@ public DruidCoordinatorRuntimeParams run(DruidCoordinatorRuntimeParams params)
           }
         }
 
+        // Skip all the locked intervals
+        LOG.debug(
+            "Skipping the following intervals for Compaction as they are currently locked: %s",
+            taskToLockedIntervals
+        );
+        taskToLockedIntervals.forEach(
```

Comment:
I think it should behave differently depending on what `lockGranularity` is used. If both the compaction task to run and the task that is already running use the segment lock, the compaction task can safely run. Otherwise, the entire locked interval should be skipped as what this code does.

Reply:
For simplicity, we are treating Segment Locks the same as Time Chunk Locks i.e. the whole interval would be skipped while submitting compaction tasks even if there is just one Segment in that interval that is locked by a higher priority task.

Added this as a javadoc comment here:
https://github.com/apache/druid/blob/a8e30c3c8c9c2b58b0615f3f50d863c334cadd75/server/src/main/java/org/apache/druid/server/coordinator/duty/CompactSegments.java#L233

https://github.com/apache/druid/blob/a8e30c3c8c9c2b58b0615f3f50d863c334cadd75/indexing-service/src/main/java/org/apache/druid/indexing/overlord/TaskLockbox.java#L690